app/components/admin/edition_images/lead_image_component.rb

```diff
@@ -37,36 +39,28 @@ def alt_text
     lead_image.alt_text.presence || "None"
   end
 
-  def new_image_display_option
-    @new_image_display_option ||= if image_display_option_is_no_image? && edition_has_images?
-                                    "custom_image"
-                                  elsif image_display_option_is_no_image?
-                                    "organisation_image"
-                                  else
-                                    "no_image"
-                                  end
-  end
+  def default_organisation_image
+    return if edition.lead_organisations.blank?
 
-  def image_display_option_is_no_image?
-    edition.image_display_option == "no_image"
+    edition.lead_organisations.first.default_news_image
   end
 
-  def update_image_display_option_button_text
-    return image_display_option_button_text_when_image_has_been_uploaded if edition_has_images?
-
-    image_display_option_button_text_when_no_images_uploaded
+  def show_default_lead_image?
+    case_study? &&
+      default_organisation_image.present? &&
+      [nil, "organisation_image"].include?(edition.image_display_option)
```

Comment:
I've seen code like this checking that the image display option is "x" or included in "[x,y]" in a few places. Feels like we might be missing an abstraction? Not sure exactly what it should be, maybe a value object for the image display option 🤔 probably not worth expending too much effort on if there's not an obvious refactor though

Reply:
Yea this is an interesting one. There's def some weird behaviour here that can probably be extracted into methods so the behaviour is more explicit. It's basically an enum that can be

`no_image`. `organisation_image`, `custom_image` or `nil`.

and combinations of them do specific things like:

`organisation_image || nil`  means the default image is passed down by the presenter to Publishing API and is rendered on the frontend.

But there's other pairings that make sense in other places. E.g. when you update the lead_image association after you toggle from `custom_image` to  `%w[no_image organisation_image]` you should delete the `edition_lead_image`.

It sort of feels like with the current implementation it should be set to `organisation_image` on create since that's the behaviour, but as it's a shared attr in the db across all editions that's not really an option.

Some other ideas would be:

1. lead image could be polymorphic and be an edition image, default_news image or speakers image (seems unlikely due to the variety of implementations of images)
2. lead_image_preferences could be it's own model and case studies could `have_one`

Tbh though it really feels like we really should give images a proper look and challenge some assumptions around implementation. Having a bunch of different subtypes editions behaving slightly differently and often having functionality being overwritten by the rendering application causes so many headaches when you work on it for what seems like pretty limited value.

It might well be useful to be able to toggle default image visibility for specific types of edition. But some having lead images, others not, some having the ability to toggle between a default image and others not feels a bit weird. Especially as we've had people request lead images for other content types.

Probs beyond the scope of this bit of work though.

Reply:
Fully agree about trying to come to a logical solution that works across content types. One to take to Darren, I think. Thanks for your response, some really good thoughts there.

I did sort of wonder also about whether trying to select an image to be a lead image, we just had a completely separate field for lead image, and whether that would make a lot of our problems go away. Probably a bit late for that though and I suspect that's going to cascade changes all the way through publishing API and schemas and all the rest of it.